proto-lens-protoc/src/Data/ProtoLens/Compiler/Generate/Encoding.hs

```diff
@@ -74,28 +77,66 @@ generatedParser m =
         | Just g <- groupFieldNumber m = do'
             [ tag <-- getVarInt'
             , stmt $ case' tag $
-                (pLitInt (groupEndTag g) --> finish exprs)
+                (pLitInt (groupEndTag g) --> finish m exprs)
                     : parseTagCases continue exprs m
             ]
         {- Regular message type:
               do
                 end <- atEnd
                 if end
-                    then return $ {reverseRepeatedFields} x
+                    then {finish}
                     else do
                         tag <- getVarInt
                         case tag of ...
         -}
         | otherwise = do'
             [ end <-- "Data.ProtoLens.Encoding.Bytes.atEnd"
             , stmt $
-                if' end (finish exprs)
+                if' end (finish m exprs)
                     $ do'
                         [ tag <-- getVarInt'
                         , stmt $ case' tag $ parseTagCases continue exprs m
                         ]
             ]
 
+{- | A Parser expression that finalizes the message.
+
+do
+    frozen'a <- unsafeLiftIO $ unsafeFreeze mutable'a
+    frozen'b <- unsafeLiftIO $ unsafeFreeze mutable'b
+    ...
+    {checkMissingFields}
+    over unknownFields reverse
+        $ set vector'a frozen'a
+        $ set vector'b frozen'b
+        ...
+        $ {partialMessage}
+-}
+finish :: MessageInfo Name -> ParseState Exp -> Exp
+finish m s = do' $
+    [ pVar frozen <-- unsafeLiftIO' @@
+                    ("Data.ProtoLens.Encoding.Growing.unsafeFreeze"
+                        @@ mutable)
+    | (frozen, mutable) <- Map.elems $ Map.intersectionWith (,)
+                                frozenNames (repeatedFieldMVectors s)
+    ]
+    ++
+    [ stmt $ checkMissingFields s
+    , stmt $ "Prelude.return" @@
+        (over' unknownFields' "Prelude.reverse"
+            @@(foldr (\(finfo, frozen) x' -> "Lens.Family2.set"
+                                            @@ lensOfVectorField finfo
```

Comment:
The 'lens' part doesn't seem to be reflected in the pseudo code above. Am I missing something?

Reply:
Good catch; I've updated the comment.